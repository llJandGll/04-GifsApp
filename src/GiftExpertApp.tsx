import React, { FormEvent, useState } from 'react'


export const GiftExpertApp : React.FC = () => {

  const [categories, setCategories] = useState(['Goku','desnudas']);


  


  return (
    <>
    {/* titulo */}
    <h1> GiftExpertApp </h1>
    {/* input */}
   
 


    
    <ol>
      {
        categories.map( ( category , id: number ) => {
          return (
            <li key={ id }>{ category }</li>
          )
        })
      }
    </ol>
    {/* listado de gifts */}

    </>
  )
}

