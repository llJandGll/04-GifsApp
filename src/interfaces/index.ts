export * from './AddCategoryProps';
export * from './CategoryProps';