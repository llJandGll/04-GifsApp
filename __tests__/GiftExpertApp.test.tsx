import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import { test, describe, expect } from 'vitest';
import { GiftExpertApp } from '../src/GiftExpertApp';


describe('test in <GiftExperApp/>', () => { 
  test('should return the aplication of title ', () => { 
    render(<GiftExpertApp/>);
    
    expect( screen.getByText('GiftExpertApp')).toBeTruthy();
  }) 
  
  test('should change the funcional component when insert input value ', () => { 
    render(<GiftExpertApp/>);

    const inputValue = 'goku';
    const input = screen.getByRole('textbox') as HTMLInputElement;
    const form = screen.getByRole('form') as HTMLFormElement;
    fireEvent.input( input , { target : {  value : inputValue } } )
    fireEvent.submit( form );

    expect( screen.getByRole('heading', { level : 2 }).textContent).toBe( inputValue )
    expect( screen.getByText('Loading...') ).toBeTruthy();
    screen.debug();
    
  })

  test('should delete a category', () => {
    render(<GiftExpertApp />);
    
    const input = screen.getByRole('textbox') as HTMLInputElement;
    const form = screen.getByRole('form') as HTMLFormElement;
    
    // Add a category
    fireEvent.input(input, { target: { value: 'goku' } });
    fireEvent.submit(form);
    
    // Delete the category
    const deleteButton = screen.getByText('Delete Gifs');
    fireEvent.click(deleteButton);
    
    expect(screen.queryByText('goku')).toBeNull();
  });

  test('should show error message for duplicate category', () => {
    render(<GiftExpertApp />);
    
    const input = screen.getByRole('textbox') as HTMLInputElement;
    const form = screen.getByRole('form') as HTMLFormElement;
    
    // Add a category
    fireEvent.input(input, { target: { value: 'goku' } });
    fireEvent.submit(form);
    
    // Try to add the same category again
    fireEvent.input(input, { target: { value: 'goku' } });
    fireEvent.submit(form);
    
    expect(screen.getByText('The category already exist')).toBeTruthy();
  });
})