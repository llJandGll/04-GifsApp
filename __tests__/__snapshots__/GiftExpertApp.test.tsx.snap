// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`test in <GiftExperApp/> > should first 1`] = `
<div>
  <h1>
     GiftExpertApp 
  </h1>
  bound HTMLFormElement {
    "0": <input
      placeholder="Buscar Gif"
      type="text"
      value=""
    />,
    "1": <button
      type="button"
    >
      Add Category
    </button>,
  }
</div>
`;

exports[`test in <GiftExperApp/> > should return the aplication of title  1`] = `
<div>
  <h1>
     GiftExpertApp 
  </h1>
  bound HTMLFormElement {
    "0": <input
      placeholder="Buscar Gif"
      type="text"
      value=""
    />,
    "1": <button
      type="button"
    >
      Add Category
    </button>,
  }
</div>
`;
